### ISSUE TITLE
Remove timeout forgiveniss for users losing to bots

### ISSUE THREAD

**Issue by Dorus** [online-go/online-go.com#2235]
First off, not 100% sure if this is the correct channel, but anyway:

Currently ranked bot games are forgiven on timeout. The idea is that a bot shouldn't lose its presumably high elo to a low ranked player on a random timeout (network issues, bot miss configuration etc).

This is all good and well, but the reverse is also set to happen now, a player can lose a ranked game with a timeout without penalty, allowing them to dodge the rank loss. Potentially playing the same bot many times until they get lucky once.

I've seen instances where players lose 10-20 ranked games in a row on timeout. The problem here is i cannot tell if this is intentionally, but the end result is they get free tries vs the bot to win a game. That shouldn't be the intention in ranked games.

Therefore, the suggestion to do the rank forgiving only when the bot has a timeout, not the player.

**Comment by anoek** [comment]
Yep, so the original logic behind that was that if someone is playing the computer, the general expectation is they don't *need* to finish out games. They can rage quit, or get bored and go play a human, or play a bot while they are waiting for a human match then leave the bot game. Especially for the later most case, when a player is playing a bot while they are waiting for a human match, I didn't want to punish them for doing that. 

However, people actively abusing that mechanic is certainly a problem. I'll ponder on that a bit more.

**Comment by Dorus** [comment]
If they want to play non-serious games, there's always the option for unranked. A recent example is 	
toposlogos88 against 12bTurboSai. I was watching these games and they whern't accidental timeouts, simply 200+ points losses that they left with the full time (5 mins, the max they allowed on the Fischer clock).

I dont think any of this is necessarily abusing anything intentionally, possibly they realize the bot doesn't care about timeouts or not, and the bot really doesn't (I do mind slightly that it blocks the bot max games, but the bot isn't that popular right now that that is an issue). But the end result is a situation that is suspicious and problematic for the ranking system.

Anyway, all that said, if somebody has a honest loss against another player/bot, we should trust the ranking system not to punch there rank down too much. If both the player and the bot are ranked correctly, a loss vs a 7 rank higher rated opponent shouldn't be too costly.

Anyway thanks for giving it some thought :)
